### PR TITLE
Ftr 4194 get data from activity log

### LINKED ISSUE #4194: Create functionality to get data from the activitylog table
Changes in DAO, Controller and REST needed to get the data from the database
-Get data per user and workspace
-Get all the user data (query for workspaces user has logs for and get them one by one)
-Create a data structure to return readable JSON with data

-Timelimits should also be added to the data loaders

### PR DESCRIPTION
**Pull request by RazgrizX**
Fixes #4194 . Adds REST points rest/activitylogs/user/{id}/workspace/{id} and rest/activitylogs/user/{id}/ to access data. Permissions set in a way, that only data about the workspaces user have permission for, will be send. Needs to be chaecked and tested later, when frontend will access data.

### CODE REVIEW

File: muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java [hunk 1]
```diff
@@ -79,7 +79,11 @@
   @Scope (PermissionScope.ENVIRONMENT)
   @DefaultEnvironmentPermissionRoles ({ EnvironmentRoleArchetype.ADMINISTRATOR, EnvironmentRoleArchetype.MANAGER, EnvironmentRoleArchetype.STUDY_PROGRAMME_LEADER, EnvironmentRoleArchetype.TEACHER, EnvironmentRoleArchetype.STUDY_GUIDER })
   public static final String ACCESS_EVALUATION = "ACCESS_EVALUATION";
-
+  
+  @Scope (PermissionScope.ENVIRONMENT)
+  @DefaultEnvironmentPermissionRoles ({ EnvironmentRoleArchetype.ADMINISTRATOR, EnvironmentRoleArchetype.MANAGER, EnvironmentRoleArchetype.STUDY_PROGRAMME_LEADER, EnvironmentRoleArchetype.TEACHER, EnvironmentRoleArchetype.STUDY_GUIDER })
```
Comment (anttiviljakainen) [inline muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java]: Is TEACHER necessary or would it be sufficient if they can access the activity log of the students at the courses they are teaching at (which has different permission)?

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/activitylog/rest/ActivityLogRESTService.java [hunk 2]
```diff
@@ -1,5 +1,121 @@
...
+  @Inject
+  private SessionController sessionController;
+  
+  @Inject
+  private WorkspaceEntityController workspaceEntityController;
+  
+  @Inject
+  private UserEntityController userEntityController;
+  
+  //TODO permissions? which ones?
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/activitylog/rest/ActivityLogRESTService.java]: Permission checking seems fine.

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/activitylog/dao/ActivityLogDAO.java [hunk 3]
```diff
@@ -20,4 +28,40 @@ public ActivityLog create(Long userEntityId, ActivityLogType type, Long workspac
...
+    if(workspaceEntityId == null)
+      predicates.add(criteriaBuilder.isNull(root.get(ActivityLog_.workspaceEntityId)));
+    else
+      predicates.add(criteriaBuilder.equal(root.get(ActivityLog_.workspaceEntityId), workspaceEntityId));
+    criteria.where(criteriaBuilder.and(predicates.toArray(new Predicate[0])));
+    criteria.orderBy(criteriaBuilder.asc(root.get(ActivityLog_.timestamp)));
+    return entityManager.createQuery(criteria).getResultList();
+  }
+  
+  //TODO: Add date parameters?
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/activitylog/dao/ActivityLogDAO.java]: Clean comments

File: muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java (hunk 1 above)
Comment (RazgrizX) [reply muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java]: @anttiviljakainen This needs additional grooming, especially the question of which courses in guider -> current student do load from a teacher perspective. If only those he teach, then ok, but if all of them are loaded, it is more complicated. Should muikku provide empty data for other courses, or not show chart icon at all? or there should be data on all the courses for teachers?

Comment (RazgrizX) [reply muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java]: @anttiviljakainen This needs additional grooming, especially the question of which courses in guider -> current student do load from a teacher perspective. If only those he teach, then ok, but if all of them are loaded, it is more complicated. Should muikku provide empty data for other courses, or not show chart icon at all? or there should be data on all the courses for teachers?

Comment (RazgrizX) [reply muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java]: @anttiviljakainen This needs additional grooming, especially the question of which courses in guider -> current student do load from a teacher perspective. If only those he teach, then ok, but if all of them are loaded, it is more complicated. Should muikku provide empty data for other courses, or not show chart icon at all? or there should be data on all the courses for teachers?

Comment (RazgrizX) [reply muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java]: This needs additional grooming, especially the question of which courses in guider -> current student do load from a teacher perspective. If only those he teach, then ok, but if all of them are loaded, it is more complicated. Should muikku provide empty data for other courses, or not show chart icon at all? or there should be data on all the courses for teachers?

Comment (RazgrizX) [reply muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java]: This needs additional grooming, especially the question of which courses in guider -> current student do load from a teacher perspective. If only those he teach, then ok, but if all of them are loaded, it is more complicated. Should muikku provide empty data for other courses, or not show chart icon at all? or there should be data on all the courses for teachers?

Comment (RazgrizX) [reply muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java]: This needs additional grooming, especially the question of which courses in guider -> current student do load from a teacher perspective. If only those he teach, then ok, but if all of them are loaded, it is more complicated. Should muikku provide empty data for other courses, or not show chart icon at all? or there should be data on all the courses for teachers?

Comment (RazgrizX) [reply muikku-core/src/main/java/fi/otavanopisto/muikku/security/MuikkuPermissions.java]: Decided to go with teacher environment permission for now. though the methods for workspace permissions are kept in place

**Pull request merged**